docs/contributing/index.rst

```diff
@@ -25,32 +38,87 @@ Please visit our `Developer Guidelines`_ if you'd like to contribute code change
 .. _`Developer Guidelines`: https://www.zope.dev/developer/guidelines.html
 .. _`guidelines for reporting bugs`: https://www.zope.dev/developer/reporting-bugs.html
 
+.. _Preperations for Contributing::
+
 Preperations for Contributing
 +++++++++++++++++++++++++++++
 
-If you want to contribute to this package, please prapare an development entvironment, that includes ``git``, ``tox`` and a posibility to have several Python Versions avaliable, for example ``pyenv``.
-Please read the part of `ref`Understanding How RestrictedPython works internally` first.
+If you want to contribute to this package, please prepare a development environment that includes ``git``, ``tox``, and several Python versions available through a Python manager such as ``pyenv``.
+Please read the part of `Understanding How RestrictedPython works internally`_ first.
```

Comment:
This will need to change to an explicit reference, instead of implicit, to align with the renamed referenced label.

```suggestion
Please read the section :ref:`Understanding-How-RestrictedPython-works-internally` first.
```